### PR TITLE
Added FilterCondition class

### PR DESCRIPTION
**Pull request by CurtisPhan**

### CODE REVIEW

File: src/data/FilterCondition.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import shlex
+from MetaDataItem import MetaDataItem
+
+def is_string(token):
+    return (token[0] == '"' and token[-1] == '"') or (token[0] == "'" and token[-1] == "'")
```
Comment (Ercadio) [inline src/data/FilterCondition.py]: This would return true for `"asdf"asd"`

File: src/data/FilterCondition.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        '>=',
+        '<',
+        '<=',
+        'and',
+        'or',
+        '(',
+        ')'
+    }
+    
+    VALID_ATTRS = MetaDataItem.attributes()
```
Comment (Ercadio) [inline src/data/FilterCondition.py]: Unlike Java, we don't need a static function to initialize a static variable. So we could do this directly
```suggestion
VALID_ATTRS = {
            'title': str,
            'url' : str,
            'collision_type' : str,
            'description' : str,
            'location' : str
        }
```

**Comment by nicholas-geofroy** [comment]
Could you write some basic tests for this class? I feel this is one of the classes that should definitely have tests

File: src/data/FilterCondition.py (hunk 2 above)
Comment (jackwilliamson) [reply src/data/FilterCondition.py]: We would need this attributes function in the FirebaseAccessor class as well, so I believe there is enough reason to maintain this static function.

File: src/data/FilterCondition.py (hunk 1 above)
Comment (CurtisPhan) [reply src/data/FilterCondition.py]: Should be fixed now. The fix will also allow escaped quotes in strings, though the tokenizer is not currently set up to detect escaped characters.

File: src/data/FilterCondition.py (hunk 2 above)
Comment (CurtisPhan) [reply src/data/FilterCondition.py]: The attributes function in MetaDataItem could be converted to a static variable instead while still being usable by other classes, but having it be a function does provide one benefit. Other classes can't modify MetaDataItem's attribute list if it's returned this way. They can change their own copy, but it won't change what gets returned by attributes().

**Pull request merged**